samza-core/src/main/java/org/apache/samza/zk/ZkLeaderElector.java

```diff
@@ -50,25 +50,30 @@
   private final String hostName;
 
   private AtomicBoolean isLeader = new AtomicBoolean(false);
-  private final IZkDataListener zkLeaderElectionListener;
+  private final IZkDataListener zkLeaderElectionInternalListener;
+  ZkLeaderElectorListener zkLeaderElectorExtListener;
   private String currentSubscription = null;
   private final Random random = new Random();
 
   @VisibleForTesting
-  ZkLeaderElector(String processorIdStr, ZkUtils zkUtils, IZkDataListener leaderElectionListener) {
+  ZkLeaderElector(String processorIdStr,
+      ZkUtils zkUtils,
+      ZkLeaderElectorListener zkLeaderElectorExtListener,
```

Comment:
One is needed for callback to the caller of the ZkLeaderElector, the other one (internal) is needed internally by this class to subscribe for previous processor changes. I will rename them to make it clearer.